README.md

```diff
@@ -34,16 +36,342 @@ version can be found on in the maven repository for
+     .name("requests_latency_seconds").help("Request latency in seconds.").register();
+
+  void processRequest(Request req) {
+    Histogram.Timer requestTimer = requestLatency.startTimer();
+    try {
+      // Your code here.
+    } finally {
+      requestTimer.observeDuration();
+    }
+  }
+}
+```
+
+The default buckets are intended to cover a typical web/rpc request from milliseconds to seconds.
+They can be overridden with the `buckets()` method on the [Histogram.Builder](https://prometheus.io/client_java/io/prometheus/client/Histogram.Builder.html#buckets-double...-).
+
+There are utilities for timing code:
+
+```java
+class YourClass {
+  static final Histogram requestLatency = Histogram.build()
+     .name("requests_latency_seconds").help("Request latency in seconds.").register();
+  
+  void processRequest(Request req) {
+    requestLatency.timer(new Runnable() {
+      public abstract void run() {
+        // Your code here.    
+      }
+    });  
+      
+      
+    // Or the Java 8 lambda equivalent  
+    requestLatency.time(() -> {
+      // Your code here.  
+    });
+  }
+}
+```
+
+### Labels
+
+All metrics can have labels, allowing grouping of related time series.
+
+See the best practices on [naming](http://prometheus.io/docs/practices/naming/)
+and [labels](http://prometheus.io/docs/practices/instrumentation/#use-labels).
+
+Taking a counter as an example:
+
+```java
+class YourClass {
+  static final Counter requests = Counter.build()
+     .name("requests_total").help("Total requests.")
+     .labelNames("method").register();
+  
+  void processGetRequest() {
+    requests.labels("get").inc();
+    // Your code here.
+  }
+}
+```
+
+### Included Collectors
+
+The Java client includes collectors for garbage collection, memory pools, JMX, classloading, and thread counts.
+These can be added individually or just use the `DefaultExports` to conveniently register them. 
+
+```java
+DefaultExports.initialize();
+```
+
+####Logging
+
+There are logging collectors for log4j and logback.
+
+To register the Logback collector can be added to the root level like so:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+    <include resource="org/springframework/boot/logging/logback/base.xml"/>
+
+    <appender name="METRICS" class="io.prometheus.client.logback.InstrumentedAppender" />
+
+    <root level="INFO">
+        <appender-ref ref="METRICS" />
+    </root>
+
+</configuration>
+```
+
+## Exporting
+
+There are several options for exporting metrics.
+
+### HTTP
+
+Metrics are usually exposed over HTTP, to be read by the Prometheus server.
+
+There are Servlet, SpringBoot, and Vert.x integrations included in the client library.
+
+To add Prometheus exposition to an existing HTTP server, see the `MetricsServlet`. 
+It also serves as a simple example of how to write a custom endpoint.
+
+```java
+context.addServlet(new ServletHolder(new MetricsServlet()), "/metrics");
```

Comment:
I mean the few lines of code you need with that to create the server and context.